### PR TITLE
Da single source tabs

### PR DESCRIPTION
**Pull request by danamansana**
Up to now there have been several sources of responsibility for tracking
tab names and which tab is selected. This commit continues the process
of reducing this problem by 1) eliminating use of tab indices in favor of using tab names 2) eliminating code that duplicated the list of tab names.

Not sure that we should merge this just yet but it could be the basis of further changes

### CODE REVIEW

**Comment by ktp242** [comment]
Thanks for doing this! I think first step we can make sure we lint the updates before we make a PR, and also write comments for the functions!

**Comment by danamansana** [comment]
I've added descriptive comments for the new functions and fixed most of the linter errors for the files changed by this branch. There are still a few linter issues in the Results component

File: src/app/components/TabItem/TabItem.jsx [hunk 1]
```diff
@@ -151,46 +126,33 @@ class TabItem extends React.Component {
...
-
-      if (tabNumber) {
-        tabIndexAttribute = (parseInt(tabNumber, 10) === j) ? null : '-1';
-      } else {
-        tabIndexAttribute = '0';
-      }
-
-      const option = (
+    tabArray.forEach((tab) => {
+      const name = getNameForFacet(tab.value);
```
Comment (ktp242) [inline src/app/components/TabItem/TabItem.jsx]: The only function of getNameForFacet from TabIndex.js is to convert the input to be a string. However, `tab.value` is a string already.  We probably don't need this function.

Comment (danamansana) [reply src/app/components/TabItem/TabItem.jsx]: Yeah this function is a holdover from when this method was more complicated. I'll remove it

**Comment by ktp242** [comment]
I think I can approve this PR! Just add the update to the changelog plz!

**Pull request merged**